Persephone/MPDClient/Extensions/MPDClient+Songs.swift

```diff
@@ -25,4 +42,51 @@ extension MPDClient {
 
     return songs
   }
+  
+  func sendFetchAlbumArt(
+    forUri songUri: String,
+    imageData: Data?,
+    offset: Int32,
+    callback: @escaping (Data?) -> Void
+  ) -> Void {
+    var size: Int?
+    
+    mpd_send_albumart(self.connection, songUri, String(offset))
+    
+    guard let sizePair = mpd_recv_pair(self.connection) else {
+      mpd_connection_clear_error(self.connection)
+      return
+    }
+    size = Int(MPDPair(sizePair).value)
+    mpd_return_pair(self.connection, sizePair)
+
+    var data = imageData ?? Data(count: size!)
+          
+    let binaryPair = MPDPair(mpd_recv_pair(self.connection))
+    let chunkSize = Int(binaryPair.value)!
+    mpd_return_pair(self.connection, binaryPair.pair)
+    
+    _ = data[offset...].withUnsafeMutableBytes { (pointer) in
+      mpd_recv_binary(self.connection, pointer, chunkSize)
+    }
+    
+    guard mpd_response_finish(self.connection) else { return }
+    
+    let newOffset = offset + Int32(chunkSize)
+    
+    if newOffset < size! {
+      DispatchQueue.main.async {
+        self.fetchAlbumArt(
+          songUri: songUri,
+          imageData: data,
+          offset: newOffset,
+          callback: callback
+        )
+      }
+    } else {
+      DispatchQueue.main.async {
+        callback(data)
+      }
+    }
```

Comment:
Since both branches of the `if` block run their code in an `DispatchQueue.main.async` block, what would you think about this structure?

```
DispatchQueue.main.async {
  if newOffset < size! {
    ...
  } else {
    ...
  }
}
```

Reply:
Turns out this did not need to be scheduled on the main thread at all, and doing so was causing problems!